### PR TITLE
Introduce poi2mimir

### PR DESCRIPTION
**Pull request by riendegris**
Create a binany poi2mimir to import point of interest in elasticsearch.

Add some tests to check data is correctly imported.

During the import, poi are attached, when available, to the closest
address, and inherit the address and admins of that place.

### CODE REVIEW

**Comment by riendegris** [comment]
This PR is 'do_not_merge' until navitia_poi_model is merged

File: src/bin/poi2mimir.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+    let model = NavitiaModel::try_from_path(file)?;
+    let poi_types = model.poi_types.clone();
+
+    // Note: We're ignoring those POIs that fail to be enriched.
+    let pois: Vec<_> = model
+        .pois
+        .into_iter()
+        .map(|(_, poi)| enrich_poi(poi, &poi_types, rubber))
+        .filter_map(|rp| rp.map_err(|err| warn!("Could not add POI: {}", err)).ok())
```
Comment (antoine-de) [inline src/bin/poi2mimir.rs]: wouldn't it be nice to have the id of the poi ?

File: src/bin/poi2mimir.rs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+lazy_static! {
+    static ref DEFAULT_NB_THREADS: String = num_cpus::get().to_string();
+}
+
+// This function takes a Poi from the navitia model, ie from the CSV deserialization, and returns
+// a Poi from the mimir model, with all the contextual information added.
+fn enrich_poi(
```
[the remark this answers is not in the document]
Comment (riendegris) [reply src/bin/poi2mimir.rs]: `into_mimir_poi`

File: src/bin/poi2mimir.rs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        labels::format_poi_label(&poi.name, admins.iter().map(|a| a.deref()), &country_codes);
+
+    let poi = Poi {
+        id: poi.id,
+        label: label,
+        name: poi.name,
+        coord: coord.clone(),
+        approx_coord: Some(coord.into()),
+        administrative_regions: admins,
+        weight: 0.0,
```
[the remark this answers is not in the document]
Comment (riendegris) [reply src/bin/poi2mimir.rs]: taking the weight of the city admin

File: src/bin/poi2mimir.rs (hunk 1 above)
Comment (riendegris) [reply src/bin/poi2mimir.rs]: :+1:

**Comment by antoine-de** [comment]
@riendegris you shouldn't merge the master into a branch to update it, you should rebase it

File: src/bin/poi2mimir.rs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    /// Number of threads to use
+    #[structopt(
+        short = "t",
+        long = "nb-threads",
+        raw(default_value = "&DEFAULT_NB_THREADS")
+    )]
+    nb_threads: usize,
+
+    /// Number of shards for the es index
+    #[structopt(short = "s", long = "nb-shards", default_value = "5")]
```
Comment (kinnou02) [inline src/bin/poi2mimir.rs]:
```suggestion
    #[structopt(short = "s", long = "nb-shards", default_value = "1")]
```
for less than 1 millions doc 1 shard is enough, most of the time we won't have 1 million POI per dataset

File: src/bin/poi2mimir.rs [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+        })
+        .collect(); // TODO Can we get rid of collect, and chain with the following rubber...?
+
+    let count = rubber
+        .bulk_index(&index, pois.into_iter())
+        .map_err(|err| format_err!("Failed bulk insertion {}", err))?;
+
+    info!("importing POIs: {} POIs added.", count);
+
+    rubber
```
Comment (kinnou02) [inline src/bin/poi2mimir.rs]: minor: I would have put this in `index_pois()` to have it close to `make_index()`

File: src/bin/poi2mimir.rs [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    let coord = Coord::from(&poi.coord);
+
+    let place = rubber
+        .get_address(&coord, None) // No timeout
+        .ok()
+        .and_then(|addrs| addrs.into_iter().next()); // Take the first place
+
+    let addr = place.as_ref().and_then(|addr| addr.address());
+
+    let admins = place.map_or_else(|| Vec::new(), |addr| addr.admins());
```
Comment (kinnou02) [inline src/bin/poi2mimir.rs]: do we really want to index poi that don't have any admins?

File: src/bin/poi2mimir.rs (hunk 5 above)
Comment (riendegris) [reply src/bin/poi2mimir.rs]: :+1:

File: src/bin/poi2mimir.rs (hunk 6 above)
Comment (riendegris) [reply src/bin/poi2mimir.rs]: :+1:

**Pull request merged**